### ISSUE TITLE
Multi-location skills – one and only one should be selected

### ISSUE THREAD

**Issue by pierrethiriet** [reinterpretcat/vrp#20]
I have a specific use case that seems not possible to model with *vrp-cli*. But I may have missed something or thought about it in the wrong way.  
Would it be possible to have a task, a *job skill*, assigned to a vehicle, that can be done in several places but only one should be selected at the end. It is similar to a location problem and thus somehow similar to dispatch or reloads, but without capacity question.  

A current example of this problem is the integration of the *refueling of the vehicles*.  
This task has to be done during the *vehicle shift*. It has to be done, but *only once* and only in *few specific petrol stations* (for accounting question). Therefore, the location/allocation of the petrol station should be done to a vehicle and integrated into its tours.  
  
I thought first to loop over petrol stations, solve each problem and then keep the less costly one. But there may be a better option?

**Comment by reinterpretcat** [comment]
Regarding the refueling example. Would break with specified location after some period of time (or distance) work better for this purpose? Logically, you don't want to refuel your vehicle right after beginning, as it is still full, or in the end, risking to run out of fuel in the middle of tour?

Interval break after some amount of time (can be extended for distance too) is supported as experimental (unstable) feature.

**Comment by pierrethiriet** [comment]
Thanks, I didn’t realize that location and even better several locations can be set for breaks.  
In this case, I guess the location of the break is optimized regarding the rest of the tour.
I’ll try this solution as it seems to fit quite well my needs.
Thanks a lot.

**Comment by pierrethiriet** [comment]
Thanks to your advice, I am now using `breaks` to include actions related to the fleet (petrol station, etc.) that should happen during a tour but potentially in different alternative locations. It is working indeed very well.

This leads me to a side question. The solution return the following stop description
```
"jobId": "break",
"type": "break",
```
Would it be possible to add a specific _jobId_ by passing an additional and optional property in break definition?
Thanks.

**Comment by reinterpretcat** [comment]
I think it can be a `tag` property, similar one for a job. However, it would be one per break definition, not per break's location.

**Comment by pierrethiriet** [comment]
Yes, indeed, it could be one `tag` per break (like refueling). For locations, that can be hendle outside _vrp-cli_ I guess.

**Comment by reinterpretcat** [comment]
I've added a break property in the master branch. The plan is to make a new release this week, I just need to add some extra checks for solution checker

**Comment by pierrethiriet** [comment]
Thanks a lot.

**Comment by pierrethiriet** [comment]
Thanks a lot.
I am not 100% sure about how passing the parameter `skip_intersection_check` in javascript context, but I'll try.

**Comment by pierrethiriet** [comment]
Sorry for my late reply,
I just comment the line `check_e1303_vehicle_breaks_time_is_correct(ctx)` :
```rust
pub fn validate_vehicles(ctx: &ValidationContext) -> Result<(), Vec<FormatError>> {
    combine_error_results(&[
        check_e1300_no_vehicle_types_with_duplicate_type_ids(ctx),
        check_e1301_no_vehicle_types_with_duplicate_ids(ctx),
        check_e1302_vehicle_shift_time(ctx),
        // check_e1303_vehicle_breaks_time_is_correct(ctx),
        check_e1304_vehicle_reload_time_is_correct(ctx),
        check_e1305_vehicle_limit_area_is_correct(ctx),
        check_e1306_vehicle_dispatch_is_correct(ctx),
        check_e1307_vehicle_has_no_zero_costs(ctx),
    ])
}
```
It indeed work as expected !
Thanks a lot.

**Comment by pierrethiriet** [comment]
I am currently updating my application with the latest version of _vrp_cli_ (v.1.18.1), and it seems that the `tag` for `vehicle shift breaks` is not passed to the solution.
In a previous version, the tag was available under `jobTag`. Is there some change in the pragmatic format? But looking at the doc, it seems not, but I may have missed something ?
Thanks,

**Comment by reinterpretcat** [comment]
Hi,

no, there was no changes in api contract. I'll check it once I have time (just returned from vacation).

**Comment by pierrethiriet** [comment]
Sorry, it is my fault.
In a previous version, `breaks` were slightly different. 
At that time, `tag` was  just on the first level:
```json
"breaks": [
  {
    "time": [
      "2019-07-04T12:00:00Z",
      "2019-07-04T14:00:00Z"
    ],
  "duration": 3600.0,
  "locations": [
    {}
  ],
  "tag": "testTag"
}
```

This is not the case anymore, as clearly stated in our docs...
```json
"breaks": [
  {
    "time": [
      "2019-07-04T12:00:00Z",
      "2019-07-04T14:00:00Z"
    ],
    "places": [
      {
        "tag": "testTag",
        "duration": 3600.0
      }
    ]
  }
]
```
Sorry for the trouble, and thanks again
